src/libraries/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -39,19 +39,23 @@ internal struct FileStatus
         internal bool IsReadOnly(ReadOnlySpan<char> path, bool continueOnError = false)
         {
             EnsureStatInitialized(path, continueOnError);
+            return IsReadOnly(_fileStatus);
+        }
+
+        internal static bool IsReadOnly(Interop.Sys.FileStatus fileStatus)
```

Comment:
Private?

Reply:
No, it needs to be accessed by `FileSystemEntry` too, which is in the same assembly.